### ISSUE TITLE
[508]`va-on-this-page` focus and `tabindex`

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#41831]
## Description

When a link within the `va-on-this-page` web component is used, the page scrolls to the `h2` with the matching text/ID, but focus is not shifted to the header because it is missing a `tabindex="-1"` attribute. To improve screenreader usage, add a `tabindex="-1"` 

cc: @joshkimux

## Tasks

- [ ] Check and add `tabindex="-1"` to the target header to allow focus
- [ ] Add tests
- [ ] Check fix with screenreader

## Definition of done

- [ ] All tasks complete
- [ ] All tests passing

**Comment by joshkimux** [comment]
@Mottie made some modifications to the ticket upon a deeper dive into the issue!

**Comment by Mottie** [comment]
@joshkimux The code you shared that removes the `tabindex` on blur hasn't been updated in 6 years. Is it okay to remove the `tabindex`? What if it was intentionally set before for some interaction? It's something to think about.

**Comment by joshkimux** [comment]
Coming back to this, I think after our conversation with @jaredcunha we confirmed that we want to move forward with this approach 🥳

**Comment by joshkimux** [comment]
Another instance of this [caught in rated disabilities audit](https://github.com/department-of-veterans-affairs/va.gov-team/issues/43707):

https://user-images.githubusercontent.com/14154792/178285630-776a02f3-53a3-4ee5-a3d8-4b900e536b01.MP4
